### DISCUSSION TITLE
Migration steps for jersey-apache-connector to jersey-apache5-connector

### DISCUSSION THREAD

**Discussion by sandeepk-veritas** [dropwizard/dropwizard#6797] [Q&A]
I am currently trying to migrate our application from Dropwziard 2.1.x to 3.0.x beta5. I am following upgrade notes mentioned at https://github.com/dropwizard/dropwizard/blob/release/3.0.x/docs/source/manual/upgrade-notes/upgrade-notes-3_0_x.rst

We internally make use of Jersey client. So looking for migration steps for jersey-apache-connector to jersey-apache5-connector. They are missing from upgrade notes. If there is something particularly different between them, please do mention the same. I could have asked the same question in their forums, but found that the changes for introducing jersey-apache5-connector are done by member @zUniQueX of Dropwizard team, so asking it here. PR: https://github.com/eclipse-ee4j/jersey/pull/4950

I will be trying migration by changing the package names and any class/enum names and it should hopefully work.

FYI, I have inherited this code, so there are still some areas of code which I am not entirely familiar. And the area where we make use of Jersey client falls into that bucket.

Also our application supports Windows authentication, for that I could find we are setting ApacheClientProperties.CREDENTIALS_PROVIDER. 

```
CredentialsProvider cp = new WindowsCredentialsProvider(new SystemDefaultCredentialsProvider());
clientConfig.property(ApacheClientProperties.CREDENTIALS_PROVIDER, cp);
```

But with httpclient5-win has removed class `WindowsCredentialsProvider`. So wondering what to do with code during migration. I asked the question on mailing list httpclient-users@hc.apache.org. I got a feedback:

> If you look at the source code of the WindowsCredentialsProvider you
> can see that it is basically useless.

But if I comment out the code above in Dropwizard 2.1.x based application, things are failing for authentication. So I am bit confused. I am yet to completely to migrate to 3.0.x beta5, so will give it a try once done.

**Comment by joschi** [comment]
https://github.com/apache/httpcomponents-client/blob/rel/v4.5.14/httpclient-win/src/main/java/org/apache/http/impl/auth/win/WindowsCredentialsProvider.java and https://github.com/apache/httpcomponents-client/blob/rel/v4.5.14/httpclient-win/src/main/java/org/apache/http/impl/auth/win/CurrentWindowsCredentials.java are indeed not doing much except for returning the current Windows user's username and an empty password.

Have you tried only using `SystemDefaultCredentialsProvider` (without `WindowsCredentialsProvider`) and does it work?

If it doesn't work you could in the worst case use the `CredentialsProvider` and the `Credentials` classes from Apache Http Client 4.5.14 which I've linked above.

**Comment by zUniQueX** [comment]
@sandeepk-veritas I'm not an expert for the Apache client and also just followed [their migration guide](https://hc.apache.org/httpcomponents-client-5.2.x/migration-guide/migration-to-classic.html).

Note on the `CredentialsProvider` that it now only allows to retrieve credentials. To also allow setting credentials, a `CredentialsStore` has to be used.
